### ISSUE TITLE
imageproc_tests fails when building offline in clean chroot

### ISSUE THREAD

**Issue by hazel-bunny** [ScanTailor-Advanced/scantailor-advanced#61]
Summary
========
When the package is built in a clean chroot and tests are run, imageproc_tests fails

Description
==========
I'm trying to package scantailor-advanced as an rpm for Fedora in [my copr](https://copr.fedorainfracloud.org/coprs/hazel-bunny/ports). The spec file for the rpm is [here](https://github.com/hazel-bunny/rpm-packaging/blob/main/packages/scantailor-advanced/scantailor-advanced.spec). For comparison, [this](https://src.fedoraproject.org/rpms/scantailor/blob/f38/f/scantailor.spec) is the spec file for good ol' scantailor. I can successfully build the rpm in a distrobox container and all the tests pass. On the other hand, when I build the rpm in either a mock chroot or in copr, scantailor-advanced builds but imageproc_tests fails.

Steps to reproduce
================
Build the package in copr or mock

Observed result
=============
https://download.copr.fedorainfracloud.org/results/hazel-bunny/ports/fedora-38-x86_64/06121112-scantailor-advanced/builder-live.log.gz

Expected result
=============
RPM should build successfully and all tests should pass

Additional information
===================
LastTest.log
```
Start testing: Jul 04 14:53 UTC
----------------------------------------------------------
1/3 Testing: math_tests
1/3 Test: math_tests
Command: "/builddir/build/BUILD/scantailor-advanced-1.0.18/build/math_tests" "--log_level=message"
Directory: /builddir/build/BUILD/scantailor-advanced-1.0.18/build/src/math/tests
"math_tests" start time: Jul 04 14:53 UTC
Output:
----------------------------------------------------------
Running 10 test cases...

*** No errors detected

<end of output>
Test time =   0.55 sec
----------------------------------------------------------
Test Passed.
"math_tests" end time: Jul 04 14:53 UTC
"math_tests" time elapsed: 00:00:00
----------------------------------------------------------

2/3 Testing: imageproc_tests
2/3 Test: imageproc_tests
Command: "/builddir/build/BUILD/scantailor-advanced-1.0.18/build/imageproc_tests" "--log_level=message"
Directory: /builddir/build/BUILD/scantailor-advanced-1.0.18/build/src/imageproc/tests
"imageproc_tests" start time: Jul 04 14:53 UTC
Output:
----------------------------------------------------------
Running 85 test cases...
Test case SeedFillTestSuite/test_gray4_random did not check any assertions
Test case SeedFillTestSuite/test_gray8_random did not check any assertions
Test case SeedFillTestSuite/test_gray_vs_binary did not check any assertions
unknown location(0): fatal error: in "SkewFinderTestSuite/test_positive_detection": signal: SIGABRT (application abort requested)
/builddir/build/BUILD/scantailor-advanced-1.0.18/src/imageproc/tests/TestSkewFinder.cpp(21): last checkpoint: "test_positive_detection" test entry

*** 1 failure is detected in the test module "Master Test Suite"

<end of output>
Test time =   1.27 sec
----------------------------------------------------------
Test Failed.
"imageproc_tests" end time: Jul 04 14:53 UTC
"imageproc_tests" time elapsed: 00:00:01
----------------------------------------------------------

3/3 Testing: core_tests
3/3 Test: core_tests
Command: "/builddir/build/BUILD/scantailor-advanced-1.0.18/build/core_tests" "--log_level=message"
Directory: /builddir/build/BUILD/scantailor-advanced-1.0.18/build/src/core/tests
"core_tests" start time: Jul 04 14:53 UTC
Output:
----------------------------------------------------------
Running 10 test cases...

*** No errors detected

<end of output>
Test time =   0.05 sec
----------------------------------------------------------
Test Passed.
"core_tests" end time: Jul 04 14:53 UTC
"core_tests" time elapsed: 00:00:00
----------------------------------------------------------

End testing: Jul 04 14:53 UTC
```

**Comment by pablogventura** [comment]
**Follow-up from maintainers**

`imageproc_tests` failing in a **clean offline chroot** is still an open packaging/reproducibility concern. Nothing in the general application merge PRs replaces a proper fix here.

**To move forward, please reply with:**
- Full **`ctest` / `imageproc_tests` output** from the failing build
- **Build environment** (distro, chroot tool, whether the network is completely blocked)
- Whether **`BUILD_TESTS=OFF`** is an acceptable workaround for your packagers in the meantime

Once we can reproduce the exact failure mode, we can target missing fixtures vs accidental network fetches in [`src/imageproc/tests`](https://github.com/ScanTailor-Advanced/scantailor-advanced/tree/master/src/imageproc/tests).

Thank you.

**Comment by pablogventura** [comment]
Opened **PR [#131](https://github.com/ScanTailor-Advanced/scantailor-advanced/pull/131)** to stop using `QPainter::drawText()` in `TestSkewFinder::test_positive_detection` (draw rotated parallel lines instead). That should eliminate the **SIGABRT in fontless mock/COPR chroots** described in the LastTest.log here.

Please re-run the RPM build/tests once that lands.

**Comment by pablogventura** [comment]
**Update:** the SkewFinder test fix was **merged** via [#131](https://github.com/ScanTailor-Advanced/scantailor-advanced/pull/131) (no `drawText` in `test_positive_detection`). Please re-run **mock/COPR** builds; if anything else fails offline, paste the new `LastTest.log` snippet.

**Comment by pablogventura** [comment]
**Resolved on `master`** via merged PR [#131](https://github.com/ScanTailor-Advanced/scantailor-advanced/pull/131) (SkewFinder test no longer uses `QPainter::drawText` in fontless chroots). If COPR/mock still fails for a **different** test, open a new issue with the new log.
